bot/exts/moderation/voice_gate.py

```diff
@@ -160,6 +185,45 @@ async def on_message(self, message: discord.Message) -> None:
         with suppress(discord.NotFound):
             await message.delete()
 
+    @Cog.listener()
+    async def on_voice_state_update(
+            self,
+            member: Member,
+            before: discord.VoiceState,
+            after: discord.VoiceState
+    ) -> None:
+        """Pings a user if they've never joined the voice chat before and aren't voice verified."""
+        if member.bot:
+            log.trace("User is a bot. Ignore.")
+            return
+
+        # member.voice will return None if the user is not in a voice channel
+        if member.voice is None:
+            log.trace("User not in a voice channel. Ignore.")
+            return
+
+        in_cache = await self.redis_cache.get(member.id, None)
+        if in_cache:
+            log.trace("User already in cache. Ignore.")
+            return
```

Comment:
This should check `if in_cache is None`. There can be the `NO_MSG` flag, which is falsey.